core/templates/dev/head/services/SpeechSynthesisChunkerService.js

```diff
@@ -0,0 +1,190 @@
+          utterance.text.substring(offset) : utterance.text);
+
+      // This regex pattern finds the next string at most 160
+      // characters in length that ends on a punctuation mark in
+      // PUNCTUATION_MARKS_TO_END_CHUNKS.
+      var delimitChunkRegex = new RegExp('^[\\s\\S]{' +
+        Math.floor(CHUNK_LENGTH / 4) + ',' + CHUNK_LENGTH + '}[' +
+        PUNCTUATION_MARKS_TO_END_CHUNKS + ']{1}|^[\\s\\S]{1,' +
+        CHUNK_LENGTH + '}$|^[\\s\\S]{1,' + CHUNK_LENGTH + '} ');
+      var chunkArray = text.match(delimitChunkRegex);
+      if (chunkArray == null ||
+          chunkArray[0] === undefined ||
+          chunkArray[0].length <= 2) {
+        // Call once all text has been spoken.
+        if (callback !== undefined) {
+          callback();
+        }
+        return;
+      }
+      var chunk = chunkArray[0];
+      newUtterance = new SpeechSynthesisUtterance(chunk);
+
+      // Copy properties from the current utterance to the next utterance,
+      // excluding the text being spoken.
+      for (var property in utterance) {
+        if (property !== 'text') {
+          newUtterance[property] = utterance[property];
+        }
+      }
+      newUtterance.onend = function() {
+        if (_speechUtteranceChunker.cancel) {
+          _speechUtteranceChunker.cancel = false;
+          return;
+        }
+        offset += chunk.length;
+        _speechUtteranceChunker(utterance, offset, callback);
+      }
+
+      // IMPORTANT!! Do not remove: Logging the object out fixes some onend
+      // firing issues. Placing the speak invocation inside a callback
+      // fixes ordering and onend issues.
+      console.log(newUtterance);
+      $timeout(function () {
+        speechSynthesis.speak(newUtterance);
+      }, 0);
+    };
+
+    var _convertToSpeakableText = function(html) {
+      elt = $('<div>' + html + '</div>');
+
+      // Convert links into speakable text by extracting the readable value.
+      elt.find('oppia-noninteractive-' + RTE_COMPONENT_NAMES.Link)
+        .replaceWith(function() {
+          if (this.attributes['text-with-value'] !== undefined) {
+            return this.attributes['text-with-value'].textContent
+              .replace(/&quot;/g, '');
+          }
+        }
+      );
+
+      // Convert LaTeX to speakable text.
+      elt.find('oppia-noninteractive-' + RTE_COMPONENT_NAMES.Math)
+        .replaceWith(function() {
+          if (this.attributes['raw_latex-with-value'] !== undefined) {
+            return this.attributes['raw_latex-with-value'].textContent
+              .replace(/&quot;/g, '')
+              // Separate consecutive characters with spaces so that 'ab'
+              // is pronounced 'a' followed by 'b'.
+              .split('').join(' ')
+              .replace(/\s*(\d+)\s*/g, '$1')
+              // Replace dashes with 'minus'.
+              .replace(/-/g, 'minus')
+              // Ensure that 'x^2' is pronounced 'x squared' rather than
+              // 'x karat 2'.
```

Comment:
karat --> caret

Reply:
oops, was recently interviewing with a company called Karat that used a ^ in their logo and this slipped out haha, fixed